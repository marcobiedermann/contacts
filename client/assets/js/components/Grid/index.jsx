import PropTypes from 'prop-types';
import React from 'react';
import styles from './style.css';

const Grid = ({ children }) => (
  <div className={styles.grid}>
    {children}
  </div>
);

Grid.propTypes = {
  children: PropTypes.node,
};

Grid.defaultProps = {
  children: null,
};

export default Grid;
