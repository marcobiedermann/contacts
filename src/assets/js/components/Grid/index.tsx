import React from 'react';
import styles from './style.module.css';

export interface GridProps {
  children: React.ReactNode;
}

function Grid(props: GridProps): JSX.Element {
  return <div className={styles.grid} {...props} />;
}

export default Grid;
