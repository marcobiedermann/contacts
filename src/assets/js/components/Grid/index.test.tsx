import { render } from '@testing-library/react';
import React from 'react';
import Grid from '.';

describe('grid component', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const { asFragment } = render(<Grid />);

    expect(asFragment).toMatchSnapshot();
  });
});
