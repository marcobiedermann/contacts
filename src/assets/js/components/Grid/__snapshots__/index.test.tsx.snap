// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grid component renders correctly 1`] = `
<div
  className="grid"
/>
`;
